ntoskrnl/mm/section.c

```diff
@@ -197,7 +197,7 @@ NTSTATUS NTAPI PeFmtCreateSection(IN CONST VOID * FileHeader,
     ULONG cbHeadersSize = 0;
     ULONG nSectionAlignment;
     ULONG nFileAlignment;
-    ULONG_PTR ImageBase;
+    ULONG_PTR ImageBase = 0;
```

Comment:
I've been contemplating this for years and never settled on a good solution.
It _feels_ like the whole function's logic is just silly and these checks like
```
if (RTL_CONTAINS_FIELD(piohOptHeader, cbOptHeaderSize, ImageBase))
    ImageBase = piohOptHeader->ImageBase;
```
should become
```
if (!RTL_CONTAINS_FIELD(piohOptHeader, cbOptHeaderSize, ImageBase))
    DIE("Invalid image....");
ImageBase = piohOptHeader->ImageBase;
```

That said, I'm okay with your change... it's better than what we have.

Reply:
I don't like the `DIE` macro at all,
but I didnt feel like refactoring this entire thing.